genie-agent/src/main/java/com/netflix/genie/agent/cli/ArgumentParser.java

```diff
@@ -0,0 +1,91 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.cli;
+
+import com.beust.jcommander.JCommander;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Command-line parser for the Genie Agent executable.
+ */
+@Slf4j
+public class ArgumentParser {
```

Comment:
I like abstracting this behind an interface so we can always swap out implementations if need be.

Reply:
Ok